test/tcheck_version.c

```diff
@@ -108,12 +112,31 @@ abort_intercept(int H5_ATTR_UNUSED sig)
     HDexit(6);
 }
 
+#ifdef H5_HAVE_WIN32_API
+/* Turns off the modal dialog that is raised when the Windows CRT calls abort().
+ *
+ * Returning TRUE here lets Windows know that we've handled the abort() and that there
+ * is no need to alert the user with a modal dialog box.
+ */
+int __cdecl
+handle_crt_abort(int reportType, char *message, int *returnValue)
+{
+    return TRUE;
+}
+#endif
+
 int
 main(int ac, char **av)
 {
+#ifdef H5_HAVE_WIN32_API
+    (void)_CrtSetReportHook2(_CRT_RPTHOOK_INSTALL, handle_crt_abort);
+#endif
```

Comment:
No need to check the return value since the test will be correct (albeit with the modal dialog box popping up) even if this call fails for some reason.